<launch>
  <!-- Launch file for Livox MID360 LiDAR -->

    <arg name="rviz" default="true" />

    <rosparam command="load" file="$(find robots_localization)/config/mid360.yaml" />

    <param name="max_iteration" type="int" value="3" />
    <param name="filter_size_surf" type="double" value="0.5" />
    <param name="filter_size_map" type="double" value="0.5" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="0" />
    <param name="map_file" type="str" value="$(find fast_lio)/PCD/yiselie360.pcd" />
    <!-- add param for manual select map pcd file -->
    <node pkg="robots_localization" type="robots_localization" name="robots_localization" output="screen" /> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find robots_localization)/rviz/localization.rviz" />
    </group>

</launch>